sphinx/source/docs/user_guide/source_examples/styling_glyph_selections.py

```diff
@@ -1,14 +1,18 @@
-from bokeh.plotting import figure, output_file, show
+from bokeh.io import output_file, show
+from bokeh.plotting import figure
+from bokeh.models import Circle
 
-output_file("axes.html")
+output_file("styling_selections.html")
 
-p = figure(plot_width=400, plot_height=400, tools="lasso_select")
-p.circle([1,2,3,4,5], [2,5,8,2,7], size=50, name="mycircle")
+p = figure(plot_width=400, plot_height=400, tools="tap", title="Select a circle")
+p.circle([1, 2, 3, 4, 5], [2, 5, 8, 2, 7], size=50, name="mycircle")
 
-glyph = p.select(name="mycircle")[0].nonselection_glyph
-glyph.fill_alpha = 0.2
-glyph.line_color = "firebrick"
-glyph.line_dash = [6, 3]
-glyph.line_width = 2
+selected_circle = Circle(fill_alpha=1, fill_color="firebrick", line_color=None)
+nonselected_circle = Circle(fill_alpha=0.2, fill_color="blue", line_color="firebrick")
+
+circle_glyph_renderer = p.select(name="mycircle")[0]
+
+circle_glyph_renderer.selection_glyph = selected_circle
+circle_glyph_renderer.nonselection_glyph = nonselected_circle
```

Comment:
This works without the `[0]` also I'd suggest just `renderer`, and a title that explicitly says to click:
```
from bokeh.io import output_file, show
from bokeh.plotting import figure
from bokeh.models import Circle

output_file("styling_selections.html")

p = figure(plot_width=400, plot_height=400, tools="tap", title="Select by Clicking")
p.circle([1, 2, 3, 4, 5], [2, 5, 8, 2, 7], size=50, name="mycircle")

selected_circle = Circle(fill_alpha=1, fill_color="firebrick", line_color=None)
nonselected_circle = Circle(fill_alpha=0.2, fill_color="blue", line_color="firebrick")

renderer = p.select(name="mycircle")

renderer.selection_glyph = selected_circle
renderer.nonselection_glyph = nonselected_circle

show(p)
```